thrift/admin.thrift

```diff
@@ -251,6 +251,11 @@ service AdminService {
     throws (
       1: shared.InternalServiceError internalServiceError,
     )
+
+  void DeleteWorkflow(1: AdminDeleteWorkflowRequest request)
+    throws (
+      1: shared.InternalServiceError internalServiceError,
```

Comment:
I haven't seen the code that depends on this, yet, but I think we should also have the `badRequestError` and `entityNotExistError` here as user errors. I'd like to reduce the number of scenarios where `internalServiceErrors` are returned so that they are always a clear signal for bugs in our system that need to be fixed.